cpu/sam0_common/periph/dac.c

```diff
@@ -110,13 +110,28 @@ int8_t dac_init(dac_t line)
 
     _dac_init_clock(line);
 
-    /* Settings can only be changed when DAC is disabled, reset config */
-    DAC->CTRLA.reg = DAC_CTRLA_SWRST;
+    /* Settings can only be changed when DAC is disabled */
+    DAC->CTRLA.reg &= ~DAC_CTRLA_ENABLE;
     _sync();
 
 #ifdef DAC_DACCTRL_ENABLE
     DAC->DACCTRL[line].reg = DAC_DACCTRL_ENABLE
-                           | _get_CCTRL(sam0_gclk_freq(DAC_CLOCK));
+                           | _get_CCTRL(sam0_gclk_freq(DAC_CLOCK))
+#endif
+#if defined(DAC_DACCTRL_RUNSTDBY) && defined(CPU_SAM0_COMMON_DAC_RUN_ON_STANDBY)
+                           | DAC_DACCTRL_RUNSTDBY
+#endif
+                           ;
+
+#ifdef DAC_DACCTRL_REFRESH
+    /** The DAC can only maintain its output on the desired value for approximately 100 μs.
+     *  For static voltages the conversion must be refreshed periodically (see e.g.
+     *  '47.6.9.3 Conversion Refresh' in the SAM D5xE5x family data sheet).
+     *
+     *  Note: T_REFRESH = REFRESH * T_OSCULP32K
+     *        (DACCTRLx.REFRESH = 1 is reserved)
+     */
+    DAC->DACCTRL[line].bit.REFRESH = 2;
```

Comment:
better make this configurable

```suggestion
    DAC->DACCTRL[line].bit.REFRESH = ONFIG_SAM0_DAC_REFRESH;
```

Reply:
Made it configurable while defaulting to 2 and adding static assertion against a value of `1`